tokio-threadpool/src/lib.rs

```diff
@@ -3,6 +3,107 @@
 #![doc(html_root_url = "https://docs.rs/tokio-threadpool/0.1.4")]
 #![deny(warnings, missing_docs, missing_debug_implementations)]
 
+// The Tokio thread pool is a thread pool designed to scheduled futures in Tokio
+// based applications. The thread pool structure manages two sets of threads:
+//
+// * Worker threads.
+// * Backup threads.
+//
+// Worker threads are primary threads and are used to schedule futures. These
+// threads operate using a work-stealing strategy. Backup threads are to support
+// the `blocking` API. Threads will transition between the two sets.
+//
+// The advantage of the work-stealing strategy is minimal cross thread
+// coordination. The thread pool attempts to make as much progress as possible
+// without communicating across threads.
+//
+// # Crate layout
+//
+// The primary type is `Pool`. This struct contains the majority of the pool
+// state. Because worker threads will shutdown and be reestarted, the pool also
```

Comment:
The primary type, `Pool`, holds the majority of a thread pool's state, including the state for each worker. Each worker's state is maintained in an instance  of `worker::Entry`.